directorywatcher/directorywatcher.go

```diff
@@ -0,0 +1,288 @@
+package directorywatcher
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"sync/atomic"
+
+	"gopkg.in/fsnotify.v1"
+
+	"github.com/reddit/baseplate.go/filewatcher"
+	"github.com/reddit/baseplate.go/internal/limitopen"
+	"github.com/reddit/baseplate.go/log"
+)
+
+// DirectoryWatcher loads and parses data from a file and watches for changes to that
+// file in order to refresh it's stored data.
+type DirectoryWatcher interface {
+	// Get returns the latest, parsed data from the DirectoryWatcher.
+	Get() interface{}
+
+	// Stop stops the DirectoryWatcher.
+	//
+	// After Stop is called you won't get any updates on the file content,
+	// but you can still call Get to get the last content before stopping.
+	//
+	// It's OK to call Stop multiple times.
+	// Calls after the first one are essentially no-op.
+	Stop()
```

Comment:
This is typically spelled `Close() error` -- types that have cleanup operations typically implement the `io.Closer` interface.